tests/lib/rules/no-undef.js

```diff
@@ -43,6 +43,7 @@ eslintTester.addRuleTest("lib/rules/no-undef", {
         "if (typeof a === 'undefined') {}",
         { code: "var toString = 1;", ecmaFeatures: { globalReturn: true }},
         { code: "function myFunc(...foo) {  return foo;}", ecmaFeatures: { restParams: true } },
+        { code: "var App; <App />;", args: [1, {vars: "all"}], ecmaFeatures: { globalReturn: true, jsx: true } },
```

Comment:
Mmm, you are right. Actually the only difference in this test is the `globalReturn` option set to `true` which trigger the bug in this case.

Maybe I can merge these two tests ?